test/server_sent_event/client_property_test.exs

```diff
@@ -0,0 +1,169 @@
+defmodule ServerSentEvent.ClientPropertyTest do
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+
+  @moduletag :property
+
+  alias ServerSentEvent.ClientTest.AutoConnect
+
+  {head, :chunked} =
+    Raxx.HTTP1.serialize_response(
+      Raxx.response(:ok)
+      |> Raxx.set_header("content-type", "text/event-stream")
+      |> Raxx.set_body(true)
+    )
+
+  @first_response head
+
+  test "the client will reconstruct sses regardless of how they get split between packets" do
+    check all sses <- StreamData.list_of(any_sse(), min_length: 1, max_length: 20),
+              split_points <- StreamData.list_of(StreamData.integer(0..10_000), max_length: 100),
+              sses_in_chunk <- StreamData.integer(1..3),
+              max_runs: 600,
+              max_run_time: 10_000 do
+      # preparing the connection
+      {port, listen_socket} = listen()
+      {:ok, client} = AutoConnect.start_link(port)
+
+      {:ok, socket} = accept(listen_socket)
+      {:ok, first_request} = :gen_tcp.recv(socket, 0, 1_000)
+      assert String.contains?(first_request, "accept: text/event-stream")
+      assert String.contains?(first_request, "\r\n\r\n")
+
+      :ok = :gen_tcp.send(socket, [@first_response])
```

Comment:
Not necessarily part of the problem but I would be tempted to add first response to the whole_response string so this test includes splitting the response head

Reply:
yeah, I was thinking the same thing.

I started doing it, but I realised the client assumes the headers are going to arrive in one packet. 

I'll add it as an issue.